source/core_http_client.c

```diff
@@ -544,6 +544,22 @@ static void processCompleteHeader( HTTPParsingContext_t * pParsingContext );
  */
 static HTTPStatus_t processHttpParserError( const http_parser * pHttpParser );
 
+/**
+ * @brief Compares at most the first n bytes of str1 and str2 without case sensitivity.
```

Comment:
What happens when `n > strlen(str1) || n > strlen(str2)`? It would be best to document that n must be less than the actual size of either string.